torch/_functorch/_aot_autograd/profile.py

```diff
@@ -0,0 +1,26 @@
+from dataclasses import dataclass
+from typing import List
+
+
+@dataclass
+class Event:
+    name: str
+    duration_sec: float
+
+
+events: List[Event] = []
+
+
+def reset() -> None:
+    global events
+    events.clear()
+
+
+def add_event(e: Event) -> None:
+    global events
+    events.append(e)
+
+
+def get_events() -> List[Event]:
+    global events
+    return events
```

Comment:
Instead of handrolling your own thing, you should use @jamesjwu's profiler, which also has integrated support with tlparse and scuba. https://github.com/pytorch/pytorch/pull/134118

Reply:
Thanks. Mostly did it for debug profiling, we can ignore this PR. I will start using James's profiler when start working on subclasses compilation perf.